java/CMakeLists.txt

```diff
@@ -103,23 +108,26 @@ if(NOT WIN32)
   #   OUTPUT_NAME ${CMAKE_CURRENT_SOURCE_DIR}/src/main/c++/jni_spark_vw_generated.h)
   # add_dependencies(javaheaders javacompile)
 
+  SET(mvn_extra_args "")
+  if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+      set(JAVA_INSTALL_PATH /usr/lib)
+      # the integration tests don't currently pass on mac
+      SET(mvn_extra_args "-DskipTests")
```

Comment:
I'm hesitant to add this without a clear path to removing it. Are the only issues to do with lib loading for the spark lib?

Reply:
Yes. I think I can fix that as well - it just needs to change the file extension.

Reply:
Are you planning on addressing this?

Reply:
@jackgerrits I would like to, yes, but not sure what my timeline can be.  If anyone else can get it done then :+1:  Just let me know if/when I would need to update this PR

Reply:
@dantswain I think we want this fixed before we merge this PR. I'd rather not let turning off tests slip by and be forgotten

Reply:
No worries.  I agree in principle.  However I'm not sure how these tests are being run currently since the integration tests don't pass on macos.  Possibly the lib is getting built some other way as part of the current CI pipeline?

Anyways, I don't mind leaving it for now, I've gotten massively derailed from working on this so it's gonna be a little while before I can come back to it.

Reply:
No prob sounds good. It's possible someone else will get to it in the meantime as well.